### PR TITLE
Allow more functionality through system METIS

### PR DESCRIPTION
**Pull request by acxz**
This is a follow up PR to #570 as when it was merged I was not available to provide my review.

There is no reason to specify the full local `metis.h` include path, with how metis is treated in cmake, `metis.h` works for both local and system metis.

See: https://github.com/borglab/gtsam/pull/570/files#diff-0106693c254359c8aceceb57d5b6c0643760cc8cce6760ccfd2d31bc6d0be2dbR26
where only metis.h is used without the ifdef.

More importantly this PR enables metis's internal api ("metislib.h") via extern C for SYSTEM_METIS. We already use the internal API for our local metis install and since we carry the header files of metis with us I do not see a problem in enabling `extern "C"` for system metis as well. The only problem that can occur with this is a desync between the local headers we carry for metis and what metis is compiled under at the system level. Considering metis is a very old and stable package that hasn't seen updates for a long time, I do not see this issue occurring and if it does should be easily fixable, by just updating our local metis copy.

Tagging @jlblancoc and @varunagrawal  for reviews. Also @berndpfrommer ig

### CODE REVIEW

**Comment by dellaert** [comment]
What specific issue are we fixing? Was there a problem that prompted this PR? Is there an issue associated with the problem?

**Comment by acxz** [comment]
> What specific issue are we fixing?

Unnecessary ifdef when including <metis.h>
FindSeparator tests not run under system metis

> Was there a problem that prompted this PR?

From my [review comment](https://github.com/borglab/gtsam/pull/570#issuecomment-729964071) in the last METIS PR:
> Even if that is changed, system metis does not provide metislib.h which means we won't be able to use the functionality of it. Thus rendering the SYSTEM_METIS flag broken. If its going to be broken when merged in, we should just not merge it in.

Currently the ifdef guard for SYSTEM_METIS just disables the associated test. We should fix that by allowing tests to run under system metis.

> Is there an issue associated with the problem?

Here is the issue: https://github.com/borglab/gtsam/issues/292#issuecomment-656276457
This PR implements @jlblancoc 's approach in https://github.com/borglab/gtsam/issues/292#issuecomment-656495923

**Comment by dellaert** [comment]
Awesome. Requesting José as a reviewer, who can merge after approving.

File: cmake/HandleMetis.cmake [hunk 1]
```diff
@@ -21,7 +21,12 @@ if(GTSAM_USE_SYSTEM_METIS)
     mark_as_advanced(METIS_LIBRARY)
 
     add_library(metis-gtsam-if INTERFACE)
-    target_include_directories(metis-gtsam-if BEFORE INTERFACE ${METIS_INCLUDE_DIR})
+    target_include_directories(metis-gtsam-if BEFORE INTERFACE ${METIS_INCLUDE_DIR}
+      # gtsam_unstable/partition/FindSeparator-inl.h uses internel metislib.h API
```
Comment (varunagrawal) [inline cmake/HandleMetis.cmake:25]: typo: internal

File: cmake/HandleMetis.cmake [hunk 2]
```diff
@@ -30,10 +35,12 @@ else()
   add_subdirectory(${GTSAM_SOURCE_DIR}/gtsam/3rdparty/metis)
 
   target_include_directories(metis-gtsam BEFORE PUBLIC
+    $<INSTALL_INTERFACE:include/gtsam/3rdparty/metis/>
     $<BUILD_INTERFACE:${GTSAM_SOURCE_DIR}/gtsam/3rdparty/metis/include>
+    # gtsam_unstable/partition/FindSeparator-inl.h uses internel metislib.h API
```
Comment (varunagrawal) [inline cmake/HandleMetis.cmake:40]: typo: internal

**Review by varunagrawal — APPROVED** [review]
Minor nits but LGTM

File: cmake/HandleMetis.cmake (hunk 1 above)
Comment (acxz) [reply cmake/HandleMetis.cmake:25]:
```suggestion
      # gtsam_unstable/partition/FindSeparator-inl.h uses internal metislib.h API
```

File: cmake/HandleMetis.cmake (hunk 2 above)
Comment (acxz) [reply cmake/HandleMetis.cmake:40]:
```suggestion
    # gtsam_unstable/partition/FindSeparator-inl.h uses internal metislib.h API
```

**Pull request merged**